std/traits.d

```diff
@@ -6524,6 +6575,40 @@ unittest
         static assert(mostNegative!T == 0);
 }
 
+/**
+Get the type that a scalar type `T` will promote to in arithmetic expressions.
```

Comment:
IIRC there was a whole five minutes in a DConf talk complaining about these rules. Is there anywhere where these are defined, and if so, could you link to it here?